source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -47,16 +47,22 @@ UdpStatsdSink::UdpStatsdSink(ThreadLocal::SlotAllocator& tls,
   });
 }
 
-void UdpStatsdSink::flushCounter(const Stats::Counter& counter, uint64_t delta) {
-  const std::string message(
-      fmt::format("{}.{}:{}|c{}", prefix_, getName(counter), delta, buildTagStr(counter.tags())));
-  tls_->getTyped<Writer>().write(message);
-}
+void UdpStatsdSink::flush(Stats::FlushDelegate& flush_delegate) {
+  Writer& writer = tls_->getTyped<Writer>();
+  for (const Stats::CounterSharedPtr& counter : flush_delegate.cachedCounters()) {
+    uint64_t delta = counter->latch();
+    if (counter->used()) {
```

Comment:
Should latch() be after `if (counter->used())`?

Reply:
I think that makes more sense.  Updated.